DevEdu.API/DevEdu.API/Controllers/CourseController.cs

```diff
@@ -11,85 +11,52 @@ namespace DevEdu.API.Controllers
     [Route("api/[controller]")]
     public class CourseController : Controller
     {
-        public CourseController()
+        private readonly IMapper _mapper;
+        private readonly ICourseRepository _courseRepository;
+        public CourseController(IMapper mapper, ICourseRepository courseRepository)
         {
+            _courseRepository = courseRepository;
+            _mapper = mapper;
         }
 
         //  api/Course/5
         [HttpGet("{id}")]
-        public string GetCourse(int id)
+        public CourseDto GetCourse(int id)
         {
-            return $"course №{id}";
+            return _courseRepository.GetCourse(id);
         }
 
         //  api/Course
         [HttpGet]
-        public string GetAllCourses()
+        public List<CourseDto> GetAllCourses()
         {
-            return "All course";
+            return _courseRepository.GetCourses();
         }
 
         //  api/course
         [HttpPost]
         public int AddCourse([FromBody] CourseInputModel model)
         {
-            return 1;
+            var dto = _mapper.Map<CourseDto>(model);
+            return _courseRepository.AddCourse(dto);
         }
 
         //  api/course/5
         [HttpDelete("{id}")]
         public void DeleteCourse(int id)
         {
-
+            _courseRepository.DeleteCourse(id);
         }
 
         //  api/course/5
         [HttpPut("{id}")]
         public string UpdateCourse(int id, [FromBody] CourseInputModel model)
         {
+            var dto = _mapper.Map<CourseDto>(model);
+            dto.Id = id;
+            _courseRepository.UpdateCourse(dto);
             return $"Course №{id} change name to {model.Name} and description to {model.Description}";
         }
 
```

Comment:
why have you deleted all the rest methods?) it might cause some conflict...
could you revert this?